src/axom/core/tests/utils_locale.hpp

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+//-----------------------------------------------------------------------------
+//
+// file: utils_locale.hpp
+// Checks locales and prints out related info
+//
+//-----------------------------------------------------------------------------
+
+#include "axom/core/utilities/StringUtilities.hpp"
+#include "axom/core/utilities/System.hpp"
+
+#include "axom/fmt.hpp"
+
+#include <locale>
+#include <string>
+#include <cstdio>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+#include <array>
+#include <vector>
+#include <set>
+#include <algorithm>
+
+#include "gtest/gtest.h"
+
+#ifdef _WIN32
+  #include <Windows.h>
+#endif
+
+namespace
+{
+#ifdef __linux__
+/**
+ * \brief Utility function to capture output of system call
+ * \note Adapted from https://stackoverflow.com/a/478960
+ */
+std::string execute_command(const std::string& cmd)
+{
+  std::array<char, 128> buffer;
+  std::string result;
+
+  std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd.c_str(), "r"), pclose);
+  if(!pipe)
+  {
+    throw std::runtime_error(
+      axom::fmt::format("popen() failed for command '{}'", cmd));
+  }
+
+  while(fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr)
+  {
+    result += buffer.data();
+  }
+
+  return result;
+}
+#endif  // __linux__
+
+#ifdef _WIN32
+/// callback function for EnumSystemLocaleEx call in enumerate_locales_windows test
+/// note: We're expecting the LPARAM to be a pointer to a vector of wide strings
+BOOL CALLBACK MyFuncLocaleEx(LPWSTR pStr, DWORD dwFlags, LPARAM strvec_ptr)
+{
+  using StrVec = std::vector<std::wstring>;
+  ((StrVec*)strvec_ptr)->push_back(pStr);
+  return TRUE;
+}
```

Comment:
@rhornung67 -- I responded to your earlier comment about why `static_cast` doesn't work here.

Here's the callback with a C-style cast instead of a `reinterpret_cast`.
Do you think this is better than the `reinterpret_cast`?

Reply:
Based on the following thread, it looks like the original `reinterpret_cast` might be better here.

See: https://stackoverflow.com/questions/7831696/reinterpret-cast-vs-c-style-cast 
and the following point in particular: https://stackoverflow.com/a/7833138